internal/engine/build_and_deployer.go

```diff
@@ -63,10 +64,12 @@ func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context,
 	var lastErr, lastUnexpectedErr error
 	logger.Get(ctx).Debugf("Building with BuildOrder: %s", composite.builders.String())
 	for i, builder := range composite.builders {
-		logger.Get(ctx).Debugf("Trying to build and deploy with %T", builder)
+		buildType := fmt.Sprintf("%T", builder)
+		logger.Get(ctx).Debugf("Trying to build and deploy with %s", buildType)
 		br, err := builder.BuildAndDeploy(ctx, st, specs, currentState)
 		if err == nil {
-			return br, err
+			span.SetAttributes(core.KeyValue{Key: "buildType", Value: core.String(buildType)})
```

Comment:
this means that if the build fails (like if there's a compiler error), you won't attach the buildType attribute. is that what you want?

Reply:
Probably not, good call.